src/main/java/org/folio/rest/impl/ProxiesForAPI.java

```diff
@@ -40,320 +36,116 @@
-                    String message = logAndSaveError(reply.cause());
-                    if(isDuplicate(message)) {
-                      Errors existsError = ValidationHelper.createValidationErrorMessage(
-                              "proxyFor", entity.getId(), "Proxy relationship already exists");
-                      asyncResultHandler.handle(Future.succeededFuture(
-                              PostProxiesforResponse.respond422WithApplicationJson(existsError)));
-                    } else {
-                      asyncResultHandler.handle(Future.succeededFuture(
-                              PostProxiesforResponse.respond500WithTextPlain(
-                                      getErrorResponse(message))));
-                    }
-                  } else {
-                    String returnObject = reply.result();
-                    entity.setId(returnObject);
-                    asyncResultHandler.handle(Future.succeededFuture(
-                      PostProxiesforResponse.respond201WithApplicationJson(entity,
-                        PostProxiesforResponse.headersFor201().withLocation(URL_PREFIX + returnObject))));
-                  }
-                } catch(Exception e) {
-                  String message = logAndSaveError(e);
-                  asyncResultHandler.handle(Future.succeededFuture(
-                              PostProxiesforResponse.respond500WithTextPlain(
-                                      getErrorResponse(message))));
-                }
-              });
-            } catch(Exception e) {
-              String message = logAndSaveError(e);
-              asyncResultHandler.handle(Future.succeededFuture(
-                      PostProxiesforResponse.respond500WithTextPlain(
-                              getErrorResponse(message))));
-            }
+    String lang,
+    ProxiesFor entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+    PostgresClient postgresClient = PostgresClientUtil.getInstance(vertxContext, okapiHeaders);
+    userAndProxyUserComboExists(entity.getUserId(), entity.getProxyUserId(),
+      postgresClient).setHandler(existsRes -> {
+        if (existsRes.failed()) {
+          String message = logAndSaveError(existsRes.cause());
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostProxiesforResponse.respond500WithTextPlain(
+              getErrorResponse(message))));
+        } else if (existsRes.result() == true) {
+          Errors existsError = ValidationHelper.createValidationErrorMessage(
+            "proxyFor", entity.getId(), "Proxy relationship already exists");
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostProxiesforResponse.respond422WithApplicationJson(existsError)));
+        } else {
+          String id = entity.getId();
+          if (id == null) {
+            id = UUID.randomUUID().toString();
+            entity.setId(id);
```

Comment:
Drop this else-clause. This is done by PgUtil.post: https://github.com/folio-org/raml-module-builder/blob/v25.0.0/domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java#L505

Reply:
nice